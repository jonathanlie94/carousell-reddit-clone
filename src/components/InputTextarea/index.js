/*
  An input textarea component, used in our create topic page.
*/

import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { theme } from 'ui';

const StyledTextarea = styled.textarea`
  font-family: Arial, sans-serif;
  border-radius: 4px;
  padding: 0.25rem;
  margin: 0;
  box-sizing: border-box;
  width: 100%;
  border-width: 2px;
  border-style: solid;
  border-color: ${props =>
    props.error ? theme.colors.yellow : theme.colors.grey};
  background-color: ${theme.colors.white};
  outline-color: ${theme.colors.blue};
  outline-width: 4px;
`;

const ErrorMessage = styled.div`
  color: ${theme.colors.red};
  margin-top: ${theme.margins.small};
`;

class InputTextarea extends PureComponent {
  render() {
    return (
      <div className={this.props.className}>
        <StyledTextarea
          cols={this.props.cols}
          rows={this.props.rows}
          ref="input"
          type="text"
          onBlur={this.props.onBlur}
          onChange={e => this.props.onChange(e.target.value)}
          value={this.props.value}
        />
        {this.props.error
          ? <ErrorMessage>
              {this.props.error}
            </ErrorMessage>
          : false}
      </div>
    );
  }
}

InputTextarea.propTypes = {
  className: PropTypes.string,
  error: PropTypes.string,
  rows: PropTypes.number,
  cols: PropTypes.number,
  value: PropTypes.string,
  onBlur: PropTypes.func,
};

InputTextarea.defaultProps = {
  rows: 20,
  cols: 80,
};

export default InputTextarea;
