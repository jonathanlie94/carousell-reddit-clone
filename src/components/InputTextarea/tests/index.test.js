import React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import 'jest-styled-components';
import InputTextarea from '../';

describe('<InputTextarea />', () => {
  it('should handle keydown correctly', () => {
    let val = 'B';
    const onChange = v => {
      val = v;
    };
    const renderedComponent = mount(
      <InputTextarea onChange={onChange} value={val} />
    );
    renderedComponent
      .ref('input')
      .simulate('change', { target: { value: 'BA' } });

    expect(val).toEqual('BA');
  });

  it('should match previous snapshot', () => {
    const renderedComponent = mount(<InputTextarea />);
    const tree = toJSON(renderedComponent);
    expect(tree).toMatchStyledComponentsSnapshot();
  });
});
