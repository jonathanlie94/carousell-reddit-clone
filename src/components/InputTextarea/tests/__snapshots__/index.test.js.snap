// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputTextarea /> should match previous snapshot 1`] = `
.jDMZsH {
  font-family: Arial, sans-serif;
  border-radius: 4px;
  padding: 0.25rem;
  margin: 0;
  box-sizing: border-box;
  width: 100%;
  border-width: 2px;
  border-style: solid;
  border-color: #C8C8C8;
  background-color: #FFF;
  outline-color: #00C3FF;
  outline-width: 4px;
}

<InputTextarea
  cols={80}
  rows={20}
>
  <div>
    <styled.textarea
      cols={80}
      onChange={[Function]}
      rows={20}
      type="text"
    >
      <textarea
        className="sc-bdVaJa jDMZsH"
        cols={80}
        onChange={[Function]}
        rows={20}
        type="text"
      />
    </styled.textarea>
  </div>
</InputTextarea>
`;
